File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = (prices) => {
    
    let left = 0;
    let right = 1;
    let maxProfit = 0;
    while (right < prices.length) {
        let currentProfit = prices[right] - prices[left]
        if (prices[left] < prices[right]) {
            maxProfit = Math.max(maxProfit, currentProfit)
        }
        else {
            left = right;
        }
        right += 1
    }
    return maxProfit;
    
};

    

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 1
        max_profit = 0
        while right < len(prices):
            current_profit = prices[right] - prices[left]
            if prices[right] > prices[left]:
                max_profit = max(current_profit, max_profit)
            else:
                left = right 
            right += 1
            
        return max_profit
    
    
    # if left is more than right, then left becomes right 
    # Input = [7,2,5,3,6,4,1,7]
    # Output = 6 (7 - 1)
    # 2 was the lowest until it hit 1 
            
        